torchmetrics/functional/text/squad.py

```diff
@@ -0,0 +1,292 @@
+
+    Raises:
+        KeyError:
+            If the required keys are missing in either predictions or targets.
+
+    References:
+        [1] SQuAD: 100,000+ Questions for Machine Comprehension of Text by Pranav Rajpurkar, Jian Zhang, Konstantin
+        Lopyrev, Percy Liang `SQuAD Metric`_ .
+    """
+
+    if isinstance(preds, Dict):
+        preds = [preds]
+
+    if isinstance(targets, Dict):
+        targets = [targets]
+
+    for pred in preds:
+        keys = pred.keys()
+        if "prediction_text" not in keys or "id" not in keys:
+            raise KeyError(
+                "Expected keys in a single prediction are 'prediction_text' and 'id'."
+                "Please make sure that 'prediction_text' maps to the answer string and 'id' maps to the key string."
+            )
+
+    for target in targets:
+        keys = target.keys()
+        if "answers" not in keys or "id" not in keys:
+            raise KeyError(
+                "Expected keys in a single target are 'answers' and 'id'."
+                "Please make sure that 'answers' maps to a `SQuAD` format dictionary and 'id' maps to the key string.\n"
+                "SQuAD Format: "
+                "{"
+                "    'answers': {"
+                "        'answer_start': [1],"
+                "        'text': ['This is a test text']"
+                "    },"
+                "    'context': 'This is a test context.',"
+                "    'id': '1',"
+                "    'question': 'Is this a test?',"
+                "    'title': 'train test'"
+                "}"
+            )
+
+        answers_keys = target["answers"].keys()
+        if "text" not in answers_keys:
+            raise KeyError(
+                "Expected keys in a 'answers' are 'text'."
+                "Please make sure that 'answer' maps to a `SQuAD` format dictionary.\n"
+                "SQuAD Format: "
+                "{"
+                "    'answers': {"
+                "        'answer_start': [1],"
+                "        'text': ['This is a test text']"
+                "    },"
+                "    'context': 'This is a test context.',"
+                "    'id': '1',"
+                "    'question': 'Is this a test?',"
+                "    'title': 'train test'"
+                "}"
+            )
+
+    preds_dict = {prediction["id"]: prediction["prediction_text"] for prediction in preds}
+    targets_dict = [
+        {
+            "paragraphs": [
+                {
+                    "qas": [
+                        {
+                            "answers": [{"text": answer_text} for answer_text in target["answers"]["text"]],
+                            "id": target["id"],
+                        }
+                        for target in targets
+                    ]
+                }
+            ]
+        }
+    ]
+    scores: Tuple[Tensor, Tensor, Tensor] = _squad_update(preds_dict, targets_dict)
+    return _squad_compute(scores)
```

Comment:
nit: keep as separate instead of passing around as tuple 
```suggestion
    f1, exact_match, total = _squad_update(preds_dict, targets_dict)
    return _squad_compute(f1, exact_match, total)
```